GCTC-NTGC/gc-digital-talent
# Feature - 3420 - Add links to nav

Resolves #3420 

## Summary

This adds new links to the navigation for:

- My profile -> Mon profile
- My applications -> Mes applications
- Browse opportunities -> Possibilités de navigation
## To Do

- [x] Add path for My applications

## Testing

1. Build the app 
   1. `npm run intl-compile`
   2. `npm run production --workspace=talentsearch`
2. As an anonymous user, navigate to `/talent`
3.  Confirm the following links
    - Search
    - Request
    - Browse opportunities
4. Login then confirm two new links appear:
   - My profile
   - My applications 

## Bonus

Added a new `useFeatureFlags` hook that returns `Record<string, boolean>` to make feature flags slightly easier to reuse in a single component.
